retrieval/discovery/kubernetes/endpoints.go

```diff
@@ -89,6 +89,11 @@ func (e *Endpoints) Run(ctx context.Context, ch chan<- []*config.TargetGroup) {
 			send(e.buildEndpoints(o.(*apiv1.Endpoints)))
 		},
 		DeleteFunc: func(o interface{}) {
+			s, ok := o.(*cache.DeletedFinalStateUnknown)
+			if ok {
+				o = s.Obj
+			}
+
 			send(&config.TargetGroup{Source: endpointsSource(o.(*apiv1.Endpoints).ObjectMeta)})
```

Comment:
I generally agree that panicking on contract violation is fine. But this is a third party dependency that's too large and too quickly changing that we can realistically validate changes and correctness on every vendoring update.

And we cannot really risk crashing our users servers just because a dependency of the SD integration is not working as we expect.